src/coreclr/jit/compiler.h

```diff
@@ -4380,10 +4380,17 @@ class Compiler
 
     GenTree* impImportLdvirtftn(GenTree* thisPtr, CORINFO_RESOLVED_TOKEN* pResolvedToken, CORINFO_CALL_INFO* pCallInfo);
 
+    enum BoxPatternFlags
+    {
+        None                  = 0,
+        IsByRefLike           = 1,
+        MakeInlineObservation = 2,
```

Comment:
These names might have a bit high chance to conflict with other names. Prefix with something like `BoxPatterns_`?

Reply:
Or use `class enum`

Reply:
I guess I could use `class enum` at this point. I was considering these more like flags, but that isn't possible with the current values. I will go with `class enum`.